tests/calculators/refl1d/test_refl1d_wrapper.py

```diff
@@ -276,3 +282,137 @@ def test_sld_profile(self):
         p.add_item('Item', 'MyModel')
         assert_almost_equal(p.sld_profile('MyModel')[1][0], 0)
         assert_almost_equal(p.sld_profile('MyModel')[1][-1], 4)
+
+
+def test_get_oversampling():
+    # When
+    q = np.linspace(1, 10, 10)
+    dq = np.linspace(0.01, 0.1, 10)
+
+    # Then
+    oversampling = _get_oversampling_q(q_array=q, dq_array=dq, oversampling_factor=5)
+
+    # Expect
+    assert len(oversampling) == 50
+    assert oversampling[0] == 0.965
+    assert oversampling[-1] == 10.35
+
+
+def test_get_probe():
+    # When
+    q = np.linspace(1, 10, 10)
+    dq = np.linspace(0.01, 0.1, 10)
+    model_name = 'model_name'
+
+    storage = {'model': {model_name: {}}}
+    storage['model'][model_name]['scale'] = 10.0
+    storage['model'][model_name]['bkg'] = 20.0
+
+    # Then
+    probe = _get_probe(q_array=q, dq_array=dq, model_name=model_name, storage=storage)
+
+    # Then
+    assert all(probe.Q == q)
+    assert all(probe.calc_Qo == q)
```

Comment:
More a question on refl1d than your code, but what is Qo? I can't work it out

Reply:
I think it is something that is needed when dealing with the more generic `Probe` rather than the `QProbe`.  

My impression was that the Q and Qo are very closely related.  Qo appears to be a "ordered" (sorted) version of Q.  For the generic `Probe` Q seems to be determined from T (incident angle) and L (incident wavelength) as a consequence the determined Q values might not be ordered.